adapters/adnuntius/adnuntius.go

```diff
@@ -369,6 +375,38 @@ func getGDPR(request *openrtb2.BidRequest) (string, string, error) {
 	return gdpr, consent, nil
 }
 
+func generateReturnExt(ad Ad, request *openrtb2.BidRequest) (json.RawMessage, error) {
+	// We always force the publisher to render
+	var adRender int8 = 0
+
+	var requestRegsExt *openrtb_ext.ExtRegs
+	if request.Regs != nil && request.Regs.Ext != nil {
+		if err := json.Unmarshal(request.Regs.Ext, &requestRegsExt); err != nil {
+
+			return nil, fmt.Errorf("Failed to parse Ext information in Adnuntius: %v", err)
+		}
+	}
+
+	if requestRegsExt != nil && requestRegsExt.DSA != nil && &ad.Advertiser != nil && &ad.Advertiser.LegalName != nil {
```

Comment:
You shouldn't need to check if `ad.Advertiser` is present via `&& &ad.Advertiser != nil` since `Advertiser` will always exist on the `Ad` struct since it isn't defined as a pointer:
```
type Ad struct {
	...
	DestinationUrls map[string]string
	Advertiser      adnAdvertiser
}
```
In the case where it isn't defined in the response from your server, when it is mapped to your structs, the `Advertiser` field will exist as a zero-value struct.